src/data-providers/dynamodb.ts

```diff
@@ -68,20 +68,29 @@ export async function updatePools(pools: Pool[]) {
   }));
 }
 
-export async function getPools(chainId?: number, lastResult?: any): Promise<Pool[]> {
+export async function getPools(chainId?: number, lastResult?: any, additionalParams?: any, useQuery = false): Promise<Pool[]> {
```

Comment:
I think it'd be better to have a `queryPools` function that uses a query and takes filter parameters rather than overloading this so much.